src/python/python-proto/python_proto/__init__.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import annotations
+
+from abc import ABCMeta, abstractmethod, abstractstaticmethod
+from typing import Any
+
+
+class SerDe(metaclass=ABCMeta):
+    @classmethod
+    def __subclasshook__(cls, subclass: Any) -> bool:
+        return (
+            hasattr(subclass, "deserialize")
+            and callable(subclass.deserialize)
+            and hasattr(subclass, "serialize")
+            and callable(subclass.serialize)
+            and hasattr(subclass, "from_proto")
+            and callable(subclass.from_proto)
+            and hasattr(subclass, "into_proto")
+            and callable(subclass.into_proto)
+        )
+
+    @staticmethod
+    @abstractstaticmethod
+    def deserialize(bytes_: bytes) -> SerDe:
+        raise NotImplementedError
+
+    @abstractmethod
+    def serialize(self) -> bytes:
+        raise NotImplementedError
+
+    @staticmethod
+    @abstractstaticmethod
+    def from_proto(proto: Any) -> SerDe:
```

Comment:
If you declare a type T you can use that here and on line 36 instead of Any.

Reply:
will fix